generator/src/generator/namespace.rs

```diff
@@ -3629,13 +3629,13 @@ impl<'ns, 'c> NamespaceGenerator<'ns, 'c> {
                 outln!(out, "let mut {} = fds.split_off(fds_len);", rust_field_name);
                 outln!(out, "std::mem::swap(fds, &mut {});", rust_field_name);
             }
-            xcbdefs::FieldDef::Expr(_) => {
-                // Only supported in requests
-                outln!(
-                    out,
-                    "// TODO: Parse {}",
-                    field.name().unwrap_or("<unnamed field>")
-                );
+            xcbdefs::FieldDef::Expr(expr_ref) => {
+                // The only expr field we ever need to parse is odd_length, which
+                // can just be ignored because it's in the minor_opcode slot,
+                // and because we can calculate how much padding we need from the
+                // string length directly.
```

Comment:
I am confused. The `QueryTextExtents` request looks like this:
```xml
  <request name="QueryTextExtents" opcode="48">
    <exprfield type="BOOL" name="odd_length">
      <op op="&amp;"><fieldref>string_len</fieldref><value>1</value></op>
    </exprfield>
    <field type="FONTABLE" name="font" />
    <list type="CHAR2B" name="string" />
[...]
```
The list has no length, so its length is inferred from the request length. However, the request length is in four byte quantities, but `CHAR2B` has size two. So, you *need* the `odd_length` field to parse this correctly.

Could you add a unit test that parses `[major, 1, [2 as u16 in native endian], 1, 2, 3, 4, 5, 6, 7, 8]` as a `QueryTextExtents` request? The resulting list should contain just `5, 6` and the `7, 8` is padding?
Or perhaps even a round-trip test that first serialises something like `QueryTextExtentsRequest { font: 0x42, string: [Char2b { byte1: 1, byte2: 2 } ]}` and then parses it again, using the `#[derive(Eq)]` for checking that "everything worked"?

Reply:
Ah, hmm. I was thinking it was for padding purposes but you're right, we need `odd_length` to know whether the last two bytes are padding or actually part of the string.

Reply:
Fixed with tests.